src/Core/src/Graphics/MauiDrawable.Android.cs

```diff
@@ -389,9 +389,13 @@ protected override void OnDraw(Shape? shape, Canvas? canvas, APaint? paint)
 
 					if (_shape != null)
 					{
-						var bounds = new Graphics.Rect(_strokeThickness / 2, _strokeThickness / 2, _width - _strokeThickness, _height - _strokeThickness);
+						var offset = _strokeThickness / 2 / _density;
+						var w = (_width - _strokeThickness) / _density;
+						var h = (_height - _strokeThickness) / _density;
```

Comment:
Calculate the cross-platform bounds by taking the platform values and dividing by the density.